pkg/config/remote/service/service.go

```diff
@@ -415,7 +415,7 @@ func NewService(cfg model.Reader, rcType, baseRawURL, hostname string, tagsGette
 		databaseFilePath = options.databaseFilePath
 	}
 	dbPath := path.Join(databaseFilePath, options.databaseFileName)
-	db, err := openCacheDB(dbPath, agentVersion, authKeys.apiKey)
+	db, err := openCacheDB(dbPath, agentVersion, authKeys.apiKey, baseURL.String())
```

Comment:
Would there ever be cause for a customer to specify a baseURL that doesn't match what we expect? I'm thinking something like a non-transparent proxy where they specify "datadog.egress.network" in config, and connections are then forwarded to datadog from that host (usually for audit purposes).

In that case, the customer changing their proxy address cause the database to be reset, but I don't think that's an issue? I'm guessing the reset is harmless in terms of safety, as it's what you're trying to do with this PR in the first place - it's not going to get stuck if the client "resets the DB on it's own" without changing the backing datadog infra right?

Reply:
Exactly, customers can use proxies.

This doesn't help if customers are changing how they resolve the hostnames without changing the url.